pages/eligibility/index.tsx

```diff
@@ -0,0 +1,375 @@
+import { NextPage } from 'next'
+import { useRouter } from 'next/router'
+import useSWR from 'swr'
+import Link from 'next/link'
+import Image from 'next/image'
+import { Tab } from '@headlessui/react'
+import { Layout } from '../../components/Layout'
+import { ComponentFactory } from '../../components/Forms/ComponentFactory'
+import { NeedHelpList } from '../../components/Layout/NeedHelpList'
+import { Input } from '../../components/Forms/Input'
+import { Alert } from '../../components/Alert'
+import ProgressBar from '../../components/ProgressBar'
+import { useState } from 'react'
+import { BenefitResult } from '../../utils/api/definitions/types'
+import { ResultKey } from '../../utils/api/definitions/enums'
+import { FAQ } from '../../components/FAQ'
+
+const dataFetcher = async (url) => {
+  const res = await fetch(url)
+  const data = await res.json()
+
+  if (res.status !== 200) {
+    throw new Error(data.message)
+  }
+  return data
+}
+
+const Eligiblity: NextPage = () => {
+  const { query } = useRouter()
+  const [oas, setOAS] = useState<BenefitResult>(null)
+  const [gis, setGIS] = useState<BenefitResult>(null)
+  const [allowance, setAllowance] = useState<BenefitResult>(null)
+  const [afs, setAFS] = useState<BenefitResult>(null)
+  const [progress, setProgress] = useState({ personal: false, legal: false })
+  const [selectedTabIndex, setSelectedTabIndex] = useState<number>(0)
+
+  // check if income is too high to participate in calculation
+  const incomeTooHigh = query && parseInt(query.income as string) > 129757
```

Comment:
really not liking defining eligibility logic here, especially that it is defined in 3 different places, at a minimum it should be broken out into a `const` so there is only one place to update it